Add Repository Index + Chart Rename + Docs Rename

ikreymer opened the pull request:
Repository Index: Generate an index.yaml in ./docx/helm-repo/index.yaml to allow for browsertrix to be a helm repository.
docs: rename docs.browsertrix.cloud -> docs.browsertrix.com
docs: update deployment doc to mention helm repo as preferred way to install
docs build action: generate repository index in GH action
publish action: update auto-generated message to mention installing from the repo.

Requires: renaming docs repo to docs.browsertrix.com to happen same time this is merged.

ikreymer committed:
helm chart: rename 'browsertrix-cloud' -> 'browsertrix'

ikreymer committed:
generate helm repo index to index.yaml as part of docs build!

ikreymer committed:
publish: update chart install to mention repo
add /repo/ placeholder dir for index.yaml

ikreymer committed:
test

ikreymer committed:
fix chart name

ikreymer committed:
readd logging for now

ikreymer committed:
docs.browsertrix.cloud -> docs.browsertrix.com

ikreymer committed:
docs: rename docs.browsertrix.cloud -> docs.browsertrix.com
more renaming, helm repo now in docs.browsertrix.com/helm-repo/
docs: recommend helm repo and installation directly from helm repo

tw4l reviewed (commented):
All looks good but I'm seeing other instances of Browsertrix Cloud in the repo still, will push a few extra commits

tw4l committed:
Remove additional instances of "Cloud" in docs and playbooks

Leave any instances in actual backend codebase and repository names
for now.

Also comments out the password-check script/CI, which is no longer
working effectively and is likely unnecessary now. Leaving to be able
to review later.

tw4l committed:
Regenerate frontend localization files

tw4l commented:
Left any instances of "cloud" where it was in the actual backend codebase or the GitHub repository name for now. Otherwise I think all the changes should be made now for README/docs/Ansible playbooks.
